src/packagedcode/bitbake.py

```diff
@@ -0,0 +1,192 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import logging
+import re
+
+from oelint_parser.cls_stash import Stash
+from packageurl import PackageURL
+
+from packagedcode import models
+
+TRACE = True
+
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+
+class BitbakeBbManifestHandler(models.DatafileHandler):
+    datasource_id = 'bitbake_bb_recipe'
+    # note that there are .bbappend, .bbclass and bitbake.conf files.
+    path_patterns = ('*.bb',)
+    default_package_type = 'bitbake'
+    description = 'BitBake bb recipe manifest'
+    documentation_url = 'https://docs.yoctoproject.org/bitbake/bitbake-user-manual/bitbake-user-manual-metadata.html'
+
+    @classmethod
+    def parse(cls, location):
+
+        oestash = Stash(quiet=True)
+
+        # add any bitbake like-file
+        # TODO: may be we should handle the bbclass and bbappend here?
+        oestash.AddFile(location)
+
+        # Resolves proper cross file dependencies
+        oestash.Finalize()
+
+        # collect all variables of interest.
+        # TODO: we should not get list values. Instead plain strings
+        data = {
+            k: ' '.join(v) if isinstance(v, (list, tuple)) else v
+            for k, v in oestash.ExpandVar(filename=location).items()
+            if v
+        }
+        name = data.get('PN')
+        version = data.get('PV')
+        description = data.get('DESCRIPTION')
+        homepage_url = data.get('HOMEPAGE')
+        download_url = data.get('PREMIRRORS')
+        declared_license = data.get('LICENSE')
```

Comment:
Please keep in mind that LICENSE settings can also be package specific (fine example can be found here https://git.yoctoproject.org/poky/plain/meta/recipes-graphics/cairo/cairo_1.16.0.bb) - so it would be better to iterate over all variable items of LICENSE and then pick the best matching, where plain LICENSE should always be the fall back